agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/detail/reminder/ReminderDetailScreen.kt

```diff
@@ -48,9 +48,19 @@ import kotlinx.datetime.LocalDateTime
 
 @Composable
 fun ReminderDetailScreenRoot(
+    type: AgendaItemType,
+    isEditing: Boolean,
     onCloseClick: () -> Unit,
     viewModel: ReminderDetailViewModel = hiltViewModel(),
 ) {
+    LaunchedEffect(isEditing) {
+        if (isEditing) {
+            viewModel.onAction(ReminderDetailAction.OnEnableEdit)
+        } else {
+            viewModel.onAction(ReminderDetailAction.OnDisableEdit)
+        }
+    }
```

Comment:
Doesn't the ViewModel already know when `isEditing` changes, since it manages the state? In that case, you wouldn't need to invoke these actions from the UI, but could directly do that from the ViewModel leading to a cleaner and more "dumb" UI

Reply:
But the `isEditing` boolean is just a normal nav arg after all, right? In that case, you could access it in the ViewModel, by passing an instance of `SavedStateHandle` to the constructor:
```
class EditViewModel(
    private val savedStateHandle: SavedStateHandle
): ViewModel() {

    private val isEditing = savedStateHandle.get<Boolean>("isEditing") ?: false
    
    var state by mutableStateOf(
        EditTextState(isEditing = isEditing)
    )
        private set
}
```